src/PowerFlowData.jl

```diff
@@ -89,6 +89,12 @@ struct PowerFlowData{
     bus_activepower_withdrawals::Matrix{Float64}
     bus_reactivepower_withdrawals::Matrix{Float64}
     bus_reactivepower_bounds::Matrix{Vector{Float64}}
+    generator_slack_participation_factors::Union{
+        Dict{String, Float64},
+        Vector{Dict{String, Float64}},
```

Comment:
I think our convention so far has been to use the same type whether it's single time period or multi time period and just have one slice along the time axis if it's single time period (see all the `Matrix` fields here). Is there a reason that doesn't work here?

Reply:
Or maybe if the interpretation of `Dict{String, Float64}` is "same factors for all time periods," you should just create a vector with the same data in all entries in the constructor? That would also save you from having so many `make_bus_slack_participation_factors` implementations

Reply:
One more option: if the only purpose of `generator_slack_participation_factors` is to allow the creation of `bus_slack_participation_factors`, do you need to store them in the struct at all?

Reply:
Thank you!

I added a possibility to just have 1 input for all time steps if the factors are same, so that not to store unnecessarily the same data multiple times.

The intended sequence is:

Generator factors -> Bus factors -> Bus active power results -> Generator active power results.

I implemented the generator factors to be stored in data for this purpose.

Reply:
OK, if the dictionaries are the same in the vector, then they are stored by reference

Reply:
Modified as suggested in a different comment: support all three variants in the ACPowerFlow kwargs, but support 2 in PowerFlowData - vector and Nothing.